### PR TITLE
feat(instance): Adding the datasource for instance ip

### PR DESCRIPTION
**Pull request by jeansebastienh**

### CODE REVIEW

File: docs/data-sources/instance_ip.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+```hcl
+# Get info by IP address
+data "scaleway_instance_ip" "my_ip" {
+  address = "0.0.0.0"
+}
+```
+
+## Argument Reference
+
+- `address` - (Required) The IP address.
```
Comment (remyleone) [inline docs/data-sources/instance_ip.md:22]: Could you add getting an IP from its ID?

Comment (jeansebastienh) [reply docs/data-sources/instance_ip.md:22]: Yes working on it

File: scaleway/data_source_instance_ip_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayInstanceServerDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `resource "scaleway_instance_ip" "ip" {}`,
+			},
+			{
+				Config: `
+					resource "scaleway_instance_ip" "ip" {}
+
```
Comment (ojamesb) [inline scaleway/data_source_instance_ip_test.go:25]:
```suggestion

```

File: scaleway/data_source_instance_ip_test.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+				Config: `resource "scaleway_instance_ip" "ip" {}`,
+			},
+			{
+				Config: `
+					resource "scaleway_instance_ip" "ip" {}
+					
+					data "scaleway_instance_ip" "ip-from-address" {
+						address = "${scaleway_instance_ip.ip.address}"
+					}
+
```
Comment (ojamesb) [inline scaleway/data_source_instance_ip_test.go:29]:
```suggestion

```

File: scaleway/data_source_instance_ip_test.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccScalewayDataSourceInstanceIP_Basic(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	ipregexp := regexp.MustCompile("(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])")
```
Comment (remyleone) [inline scaleway/data_source_instance_ip_test.go:13]: Does it work for IPv6? Should we use native parsing functions? https://pkg.go.dev/net#ParseIP ?

Comment (remyleone) [reply scaleway/data_source_instance_ip_test.go:13]: We even got some help from the validation package: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/validation#IsIPAddress

File: scaleway/data_source_instance_ip.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+	// Set 'Optional' schema elements
+	addOptionalFieldsToSchema(dsSchema, "address")
+	dsSchema["id"] = &schema.Schema{
+		Type:         schema.TypeString,
+		Optional:     true,
+		Description:  "The ID of the IP address",
+		ValidateFunc: validationUUIDWithLocality(),
+	}
+	dsSchema["address"].ConflictsWith = []string{"id"}
```
Comment (Monitob) [inline scaleway/data_source_instance_ip.go:24]: Can we treat this schema as here https://github.com/scaleway/terraform-provider-scaleway/blob/aa12d5d0b5f90d3ee5d480c53a387004c43ec7b9/scaleway/data_source_lb_ip.go#L17

Comment (jeansebastienh) [reply scaleway/data_source_instance_ip.go:24]: You mean with the `ip_` prefix ?

**Review by Monitob — APPROVED** [review]

**Pull request merged**